### PR TITLE
Added Test cases for IPV6 and NO_LINUX_NL

### PR DESCRIPTION
**Pull request by nikunjdh**
Updated automation test to add test cases for
- IPV6 Addresses 
- NO_LINUX_NL support 
Improved error handling when tests fails. 
Signed-off-by: [nsdesai@gmail.com](mailto:nsdesai@gmail.com)

### CODE REVIEW

File: tests/sonic-sw-dp-bring-down.yaml [hunk 1]
```diff
@@ -18,23 +18,23 @@
    - name: Stop SONiC VPP Container 
      shell: | 
        cd sonic-vpp
-       ./start_sonic_vpp.sh stop -n sonic-vpp
+       NO_LINUX_NL={{ no_linux_nl }} ./start_sonic_vpp.sh stop -n sonic-vpp
```
Comment (patilshashidhar) [inline tests/sonic-sw-dp-bring-down.yaml:21]: stop command does not require passing the env variable NO_LINUX_NL. You can remove the env in all the stop instances.

File: tests/sonic-vpp-vm-bring-up.yaml [hunk 2]
```diff
@@ -240,6 +230,11 @@
      failed_when: command_result.stderr != ''
      tags:
      - build      
+   - name: Pause for 2 minutes for routing to sattle
```
Comment (patilshashidhar) [inline tests/sonic-vpp-vm-bring-up.yaml:233]: Routing configuration to take effect should not take 2 minutes. Are you sure ?

File: tests/validate-sonic-vpp-docker-routing.py [hunk 3]
```diff
@@ -166,10 +175,12 @@ def runSonic(action, topology, routing, verboselogArg):
 parser.add_argument("--topology", type=str, help="enter topology - {'singleHop' | 'multiHop'}", nargs='?', default="multiHop")
 parser.add_argument("--routing", type=str, help="enter routing protocol - {'static' | 'bgp'}", nargs='?', default="bgp")
 parser.add_argument("--IPPool", type=str, help="enter IPPool - example : '172.16.0.0/16'", nargs='?', default="172.16.0.0/16") 
+parser.add_argument("--IPV6Pool", type=str, help="enter IPV6Pool - example : '2001:0db8::/64'", nargs='?', default="2001:0db8::/64") 
 parser.add_argument("--host", type=str, help="enter host - example : '10.64.98.181'", nargs='?',required=True )   
 parser.add_argument("--username", type=str, help="enter SSH username for host", nargs='?',required=True ) 
 parser.add_argument("--password", type=str, help="enter SSH user password for host", nargs='?',required=True )                       
 parser.add_argument("--verboselog", type=str, help="enter verbose logging - example : 'v or vv or vvv etc'", nargs='?', default="v")
+parser.add_argument("--noLinuxNl", type=str, help="enter noLinuxNl - example : 'yes' or 'no'", nargs='?', default="yes")
```
Comment (patilshashidhar) [inline tests/validate-sonic-vpp-docker-routing.py:183]: Can you please make the default value for noLinuxNL as "no" ? Once noLinuxNL is stable we can make default as "yes".

File: tests/sonic-sw-dp-bring-down.yaml (hunk 1 above)
Comment (nikunjdh) [reply tests/sonic-sw-dp-bring-down.yaml]: Removed.

**Comment by nikunjdh** [comment]
@patilshashidhar Resolved comments, please merge when you get a chance.

**Pull request merged**